KinectServer/KinectDevice.cpp

```diff
@@ -8,8 +11,8 @@ KinectDevice::KinectDevice():
 {
     auto validate = [this](int32_t result) {
         this->isValid = result;
-        if (FAILED(this->isValid)) {
-            throw "Initializing Kinect device failed.";
+        if (this->isValid == 0) {
```

Comment:
Yes, the result is `0`. What was your results when you had the Kinect connected or not connected?